xbmc/Application.cpp

```diff
@@ -2939,6 +2939,36 @@ void CApplication::Stop(int exitCode)
   Sleep(200);
 }
 
+bool CApplication::ResumePlayer(const CFileItemPtr& itemIn) {
+	CLog::Log(LOGDEBUG, "CApplication::ResumePlayer resume item %s", itemIn->GetPath().c_str());
+	if (m_pPlayer->HasPlayer())
+	{
+		CLog::Log(LOGWARNING, "CApplication::ResumePlayer can't resume, player is already startet");
+		return false;
+	}
+
+	CFileItem item(itemIn->GetItemToPlay());
+	CLog::Log(LOGDEBUG, "CApplication::ResumePlayer resume item to play %s", item.GetPath().c_str());
+#ifdef HAS_DVD_DRIVE
+	if (item.IsDVD() || item.IsCDDA())
+		return MEDIA_DETECT::CAutorun::PlayDisc(item.GetPath(), true, false);
+#endif
+
```

Comment:
the whole logic of this method was copied from https://github.com/xbmc/xbmc/blob/e802494b89c5a684e87d1213653d5e621bfd4a91/xbmc/video/ContextMenus.cpp#L148

Should I do a refactoring to split it into smaler methods and call the application methods by ContextMenus?